internal/literals/literals.go

```diff
@@ -224,10 +265,19 @@ func obfuscateByteArray(data []byte, length int64) *ast.CallExpr {
 				},
 			}},
 		},
-		ah.ReturnStmt(ast.NewIdent("newdata")),
 	}
 
+	retname := "newdata"
+	if ref {
+		retname = "&newdata"
```

Comment:
nit: this is abusing the printer a bit, as you're creating a Go identifier containing the prefix `&`. assuming that we want to use the AST properly, and I think we do to be safe, you'd have to use an `ast.UnaryExpr`.

Reply:
Absolutely, didn't think of this